File: benchmark/ruby/newton_sqrt.rb
def sqrt_newton(n)
    return 0 if n == 0
    x = n / 2.0
    for _ in 0...20
        x = (x + n / x) / 2.0
    end
    x
end

start = Time.now.to_f * 1000
s = 0.0
for i in 1..10000
    s += sqrt_newton(i)
end
finish = Time.now.to_f * 1000
puts "Sum: #{s}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/nodejs/newton_sqrt.js
function sqrt_newton(n) {
    if (n === 0) {
        return 0;
    }
    let x = n / 2.0;
    for (let i = 0; i < 20; i++) {
        x = (x + n / x) / 2.0;
    }
    return x;
}

const start = Date.now();
let s = 0.0;
for (let i = 1; i <= 10000; i++) {
    s += sqrt_newton(i);
}
const end = Date.now();
console.log(`Sum: ${s}`);
console.log(`Time: ${end - start}ms`);

